### PR TITLE
Fix mapping confusion, and support act field when list key has more than three elements

### PR DESCRIPTION
**Pull request by nilsga**
The es mapping script does not reflect the code that inserts data, and the `act` field in the documents does not take into consideration keys that have more than three elements (dots). This commit is an attemtp to fix this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +42,6 @@ Add a structure to your configuration called "elasticsearch"
 	 //indexTimestamp: "month", //for index statsd-2015.01
 	 indexTimestamp: "day",     //for index statsd-2015.01.01
 	 countType:     "counter",
-	 timerType:     "timer",
```
Comment (nilsga) [inline README.md]: This is just overwritten by the next line anyway.

File: es-index-template.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ curl -XPUT localhost:9200/_template/statsd-template -d '
                 }
             }
         },
-        "timer" : {
```
Comment (nilsga) [inline es-index-template.sh]: renamed according to the default configuration.

File: lib/elasticsearch.js [hunk 3]
```diff
@@ -139,11 +139,12 @@ var flush_stats = function elastic_flush(ts, metrics) {
   for (key in metrics.counters) {
     var listKeys = key.split('.');
     var value = metrics.counters[key];
+    var act = listKeys.slice(3, listKeys.length).join('-');
```
Comment (nilsga) [inline lib/elasticsearch.js]: Create the `act` field out of the "remaining" array elements.

**Comment by markkimsal** [comment]
Can you rebase this onto 0.3.0 now that the type configuration is fixed?

**Comment by markkimsal** [comment]
So you want the "action" key to be all the remaining dotted keys after 4 dotted key names?

```
accounts.authentication.password.failed.attempts:1|c
```

would be
```
{
    "_type":"counter",
    "ns":"accounts",
    "grp":"authentication",
    "tgt":"password",
    "act":"failed-attempts",
    "val":"1",
    "@timestamp":"1393853783000"
}
```

There was a discussion a while back about how to make a universal mapping for any length of dotted keys.  I was looking into using ES's directory hierarchy tokenizer but that doesn't allow you to query wildcards in the middle of keys, only at the beginning or end.

```
accounts.authentication.password.failed
```
would be tokenzied to
```
accounts, accounts.authentication, accounts.authentication.password, accounts.authentication.password.failed
```

See #1 for more details.

**Comment by nilsga** [comment]
Have merged with 0.3 now.

I'm using a metric library which encodes the `actions` as `action.subaction1`, `action.subaction2` etc. So with the current implementation, all keys are trunked to `action`, which causes multiple metrics to be "squashed" into one key, and basically makes the data unusable to us. We're planning on using Kibana as a view into the data, and we need to be able to filter these keys separate from each other.

**Comment by nilsga** [comment]
Yeah, #1 seems to be related. I don't know too much about ES analyzers, but to us it would be important that the remaining fields will be filterable within Kibana, so I suspect that a solution with explicit extra fields would be most suitable for us.

**Comment by markkimsal** [comment]
Well, the original idea was to let each project split up the key however they saw fit, and the whole namespace, group, action were just place holders.  Taking all the remaining keys and making them into the action is not a bad idea.  I just wanted to be sure you knew how this squashing would affect the data.

Perhaps a config like keyBehavior: 'squashTrailingKeys' would be a good way to add this behavior without closing the door on other behaviors.

**Comment by nilsga** [comment]
Sure, I guess I just figured this to be a better default behaviour than getting collisions, as long as there was no such option as you describe. But if you have other plans, I totally understand if you think this is not the best general solution.

**Comment by markkimsal** [comment]
It might be the best general solution.  I think it's better than truncating and getting collisions.  I think the only real question is, why concatenate with dashes instead of just reconcating with dots again?

**Comment by nilsga** [comment]
Dash was more or less a random choice. So dot is as good as anything else.

**Comment by markkimsal** [comment]
I like the changes to lib/elasticsearch.js, I'd rather combine with a dot than a dash just because it's the same character so potentially less confusion.  But the changes to the readme and the es-index-template.sh aren't necessary.  0.3.0 straightened out the naming problems that you pointed out.  If you can rewrite this pull request w/o those 2 file changes I can merge it in.  If you don't care about the PR I can just write your changes into lib/elasticsearch.js.

**Comment by nilsga** [comment]
I have updated the PR with the changes you suggested. Thanks.

**Pull request merged**